vanilla/vanilla
# Add validation for maximum post length

We aren't currently setting a limit for the amount of characters a post can contain.  The body field in the discussions table is of type text which holds a limit of approximately 65,000 characters (64kb).  A user can enter any amount in the _Max Comment Length_ field.  This could lead to some confusion as to why posts are being cut off or the user is receiving an error.  A check is also done in the utility update to see if the current limit is too high, if so we set it to an agreed limit of 50,000 characters.

Note:  We selected 50,000 characters as the limit for now, but utf8 characters will have to be considered.  A utf8 character could be up to 4bytes.  If a post consists of all utf8 characters, it could be limited to approximately 16,000 characters in a text field.  A separate issue is being created to open a discussion on how to address this (#8042)

To test:

Posting page

1) Navigate to /vanilla/settings/posting
2) In the Max Comment Length field enter a number > 50k
3) An error should appear
4) Repeat step with a number < 50k
5) Form should be saved

Utility Update

1) In you config. php file set $Configuration['Vanilla']['Comment']['MaxLength'] = 51000;
2) run utility update ie. **domain/**/dashboard/utility/update
3) Check your config, the setting should have changed to 50,000


Closes #8041
